syft/pkg/cataloger/java/archive_parser_test.go

```diff
@@ -283,9 +283,7 @@ func TestParseJar(t *testing.T) {
 				// ignore select fields (only works for the main section)
 				for _, field := range test.ignoreExtras {
 					if metadata.Manifest != nil && metadata.Manifest.Main != nil {
-						if _, ok := metadata.Manifest.Main[field]; ok {
-							delete(metadata.Manifest.Main, field)
-						}
+						delete(metadata.Manifest.Main, field)
```

Comment:
I like that this simplifies the test! Just for my own sake, why do we no longer need this check for `ok`?

Reply:
Go is smart: 
> The delete built-in function deletes the element with the specified key (m[key]) from the map. If m is nil or there is no such element, delete is a no-op.

https://pkg.go.dev/builtin#delete